attest/verifier/config/src/lib.rs

```diff
@@ -6,94 +6,137 @@
 
 extern crate alloc;
 
-use alloc::{borrow::ToOwned, collections::BTreeMap, string::String, vec::Vec};
+use alloc::{
+    collections::BTreeMap,
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
 use displaydoc::Display;
-use mc_attest_core::{MrEnclave, MrSigner};
-use mc_attest_verifier::{MrEnclaveVerifier, MrSignerVerifier, Verifier, DEBUG_ENCLAVE};
 use serde::{Deserialize, Serialize};
 
-/// Defines a json schema for an individual attestation status verifier.
-/// This is either a MRENCLAVE or MRSIGNER type, and the type is inferred by
-/// the presence of these fields.
-/// Unknown fields are flagged as an error.
+/// Trusted measurement for MRENCLAVE values.
+///
+/// The MRENCLAVE is the hash over the enclave pages loaded into the SGX
+/// protected memory. Whenever the contents of the signed enclave have changed,
+/// its MRENCLAVE will change.
+///
+/// This measurement is also referred to as "Strict Enclave Modification
+/// Policy".
+///
+/// Supports de/serialization to/from JSON. Unknown JSON fields are flagged as
+/// an error.
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
-#[serde(untagged)]
 #[serde(deny_unknown_fields)]
-pub enum StatusVerifierConfig {
-    /// A MRENCLAVE-based status verifier
-    Mrenclave {
-        /// The hex-encoded bytes of the MRENCLAVE measurement. This is
-        /// enclavehash in the .css file
-        #[serde(with = "hex", rename = "MRENCLAVE")]
-        mr_enclave: [u8; 32],
-        /// The list of config advisories that are known to be mitigated in
-        /// software at this enclave revision.
-        #[serde(default)]
-        mitigated_config_advisories: Vec<String>,
-        /// The list of hardening advisories that are known to be mitigated in
-        /// software at this enclave revision.
-        #[serde(default)]
-        mitigated_hardening_advisories: Vec<String>,
-    },
-    /// A MRSIGNER-based status verifier
-    Mrsigner {
-        /// The hex-encoded bytes of the MRSIGNER measurement. This is a digest
-        /// of the modulus in the .css file. Use a tool to see what it is.
-        #[serde(with = "hex", rename = "MRSIGNER")]
-        mr_signer: [u8; 32],
-        /// The product id that this verifier checks for.
+pub struct TrustedMrEnclaveMeasurement {
```

Comment:
As mentioned in the comment on the readme, I think the `TrustedXXXMeasurement` types will need to get moved out to a different module and the `TrustedMeasurementSet` will eventually be the only thing here for the JSON.

But I want to hold off moving out until we bring in the new types for ProductID, SVN, MRSIGNER and MRENCLAVE, these will be coming in with the SGX crates, but don't want to cross the stream with that work yet. I've added tasks to #3369 for this.